### PR TITLE
libmaxminddb 1.1.1

### PR DESCRIPTION
**Pull request by oschwald**

### CODE REVIEW

File: Library/Formula/libmaxminddb.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Libmaxminddb < Formula
   desc "C library for the MaxMind DB file format"
   homepage "https://github.com/maxmind/libmaxminddb"
-  url "https://github.com/maxmind/libmaxminddb/releases/download/1.0.4/libmaxminddb-1.0.4.tar.gz"
-  sha1 "57548d426d43b9b43c77786b08594d48d0c88c62"
+  url "https://github.com/maxmind/libmaxminddb/releases/download/1.1.1/libmaxminddb-1.1.1.tar.gz"
+  sha1 "5ab295438ad914e704d69e3636fc2d951ab97e27"
```
Comment (dunn) [inline Library/Formula/libmaxminddb.rb]: We've deprecated the use of `sha1`, could you upgrade this to the `sha256`? Thanks!

Comment (oschwald) [inline Library/Formula/libmaxminddb.rb]: Done.

**Comment by dunn** [comment]
Thanks! One more change, if you could: let's make the testfile into a resource so we can check its `sha256` as well, and also cache it.  Add a resource block above the install section:
```ruby
  resource "mmdb" do
    url "http://geolite.maxmind.com/download/geoip/database/GeoLite2-Country.mmdb.gz"
    sha256 "4819459376f61423f5fb96eff141bf486d8e6133c47acd832d2ba87a39f2a922"
  end
```
Then update the test block:
```ruby
  test do
    resource("mmdb").stage do
      system "#{bin}/mmdblookup", "-f", "GeoLite2-Country.mmdb",
                                  "-i", "8.8.8.8"
    end
  end
```

**Comment by oschwald** [comment]
That file changes monthly. I didn't write that part of the formula, but my counter suggestion would be to remove the `test` section completely. The library includes a comprehensive test suite. If the `make check` works, it is unlikely that the build failed.

**Comment by oschwald** [comment]
I went ahead and removed that test.

**Comment by DomT4** [comment]
The problem is we use our test block for more than just compile. If `geoipupdate` was ever made a mandatory dependency, every time it was updated we'd pour `libmaxminddb`'s bottle and use our `test do` block to assert there were no problems against the dependency. We can't do that with the block removed.

**Comment by oschwald** [comment]
Ah, I didn't realize that. I can change it to just do the same looked on a test database that is included with libmaxminddb.

File: Library/Formula/libmaxminddb.rb [hunk 2]
```diff
@@ -44,9 +37,7 @@ def install
   end
 
   test do
-    system "curl", "-O", "http://geolite.maxmind.com/download/geoip/database/GeoLite2-Country.mmdb.gz"
-    system "gunzip", "GeoLite2-Country.mmdb.gz"
-    system "#{bin}/mmdblookup", "-f", "GeoLite2-Country.mmdb",
-                                "-i", "8.8.8.8"
+    system "#{bin}/mmdblookup", "-f", "#{buildpath}/t/maxmind-db/test-data/GeoIP2-City-Test.mmdb",
```
Comment (DomT4) [inline Library/Formula/libmaxminddb.rb]: You'll need to install this somewhere. The `buildpath` is only present during compile, the `testpath` is only present during the test. The two are sandboxed from each other.

Something like `(share/"examples").install buildpath/"t/maxmind-db/test-data/GeoIP2-City-Test.mmdb"` should be fine, at the end of `def install`.

Comment (oschwald) [inline Library/Formula/libmaxminddb.rb]: Thanks! I did this.

**Comment by oschwald** [comment]
Ok, I think this is finally good to go. Thanks for all the review!

**Comment by dunn** [comment]
Looks great, thanks again! Merged in c39587140e81628cb87867580a460abcdfda568b.

**Pull request closed without merging**